hstream/src/HStream/Server/Exception.hs

```diff
@@ -86,6 +86,15 @@ mkExceptionHandle retFun cleanFun = flip catches [
 defaultExceptionHandle :: IO (ServerResponse 'Normal a) -> IO (ServerResponse 'Normal a)
 defaultExceptionHandle = mkExceptionHandle returnErrResp $ return ()
 
+appendStreamExceptionHandle :: IO (ServerResponse 'Normal a) -> IO (ServerResponse 'Normal a)
+appendStreamExceptionHandle action = defaultExceptionHandle $ catches action [
```

Comment:
nested catches are not recommended, maybe try to have the handler list `defaultHandlers :: [Handler]` separated from `defaultExceptionHandler` and  you can append 
```hs
[ Handler (\(err :: Store.NOTFOUND) -> do
    returnErrResp StatusUnavailable $ StatusDetails (BS.pack (show err))),
   Handler (\(err :: Store.NOTINSERVERCONFIG) -> do
    returnErrResp StatusUnavailable $ StatusDetails (BS.pack (show err)))
] 
++ defaultHandlers
```